src/ucs/datastruct/usage_tracker.c

```diff
@@ -0,0 +1,244 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "usage_tracker.h"
+
+#include <ucs/debug/log.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/sys/math.h>
+
+#define UCS_USAGE_TRACKER_VERIFY_SCORE_PARAM(_param) \
+    if (_param > 1 || _param < 0) { \
+        ucs_error("##param must be in the range [0-1] (actual=%.2f)", _param); \

```

Comment:
why ## in the beginning of the message?

Reply:
doesn't it replaced with the param name by the pre-compiler?

Reply:
not when it's part of a string. pls try and see what it really prints